### PR TITLE
Added favorite owners

### PR DESCRIPTION
**Pull request by rskrzypczak**
image

### CODE REVIEW

**Comment by TomaszQr** [comment]
Add deleting entries in u_yf_favorite_shared_owners when deleting groups

File: app/Fields/Owner.php [hunk 1]
```diff
@@ -641,22 +641,64 @@ public static function getUserLabel($id, $single = false)
...
+
+		switch (\App\Fields\Owner::getType($ownerId)) {
+			case 'Users':
+				$ownerList = $this->getAccessibleUsers('', $ownerFieldType);
+				break;
+			case 'Groups':
+				$ownerList = $this->getAccessibleGroups('', $ownerFieldType);
+				break;
+			default:
+				throw new \App\Exceptions\NoPermitted('LBL_PERMISSION_DENIED');
```
Comment (mariuszkrzaczkowski) [inline app/Fields/Owner.php]: ERR_... and code

File: app/Fields/Owner.php [hunk 2]
```diff
@@ -641,22 +641,64 @@ public static function getUserLabel($id, $single = false)
...
+		}
+		if (!isset($ownerList[$ownerId])) {
+			throw new \App\Exceptions\NoPermitted('LBL_PERMISSION_DENIED');
```
Comment (mariuszkrzaczkowski) [inline app/Fields/Owner.php]: ERR_... and code

File: modules/Vtiger/actions/Fields.php [hunk 3]
```diff
@@ -240,4 +241,28 @@ public function verifyIsHolidayDate(\App\Request $request)
...
+	 * @param \App\Request $request
+	 *
+	 * @throws \App\Exceptions\IllegalValue
+	 * @throws \App\Exceptions\NoPermitted
+	 * @throws \yii\db\Exception
+	 */
+	public function changeFavoriteOwner(\App\Request $request)
+	{
+		if (!AppConfig::module('Users', 'FAVORITE_OWNERS') || (\App\User::getCurrentUserRealId() !== \App\User::getCurrentUserId())) {
+			throw new \App\Exceptions\NoPermitted('LBL_PERMISSION_DENIED', 406);
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/actions/Fields.php]: ERR_..

**Pull request merged**